File: 卡码/代码随想录/Go/二叉树/01-二叉树遍历方式/LeetCode_0199.go
//199. 二叉树的右视
package main

import (
	. "binaryTree/types"
)

func rightSideView(root *TreeNode) []int {
	res:=make([]int,0)
	q:=make([]*TreeNode,0)
	if root!=nil {
		q=append(q,root)
	}
	var size int
	for len(q)>0 {
		size=len(q)
		for i:=0;i<size;i++ {
			node:=q[0]
			q=q[1:]
			if i==size-1 {
				res=append(res,node.Val)
			}
			if node.Left!=nil {
				q=append(q,node.Left)
			}
			if node.Right!=nil {
				q=append(q,node.Right)
			}
		}
	}
	return res
}

File: 卡码/代码随想录/JS/二叉树/01-二叉树遍历方式/LeetCode_0199.js
//199. 二叉树的右视
import TreeNode from "../TreeNode.js";

const rightSideView = function(root) {
  const res=new Array();
  const q=new Array();
  let size=0;
  if(root) q.push(root);
  while(q.length>0){
    size=q.length;
    for(let i=0;i<size;i++){
      const node=q.shift();
      if(i==size-1) res.push(node.val);
      if(node.left) q.push(node.left);
      if(node.right) q.push(node.right);
    }
  }
  return res;
};